Http/HttpRequest.cs

```diff
@@ -8,20 +8,30 @@ public class HttpRequest
 	{
 		public UnityWebRequest UnityWebRequest { get; private set; }
 
-		private readonly Dictionary<string, string> headers;
+		private Dictionary<string, string> headers;
 
 		public HttpRequest(UnityWebRequest unityWebRequest)
 		{
 			UnityWebRequest = unityWebRequest;
 
-			headers = Http.Instance.GetSuperHeaders();
+			SetDefaultHeaders();
 		}
 
 		public void SetHeader(string key, string value)
 		{
 			headers[key] = value;
 		}
 
+		public void SetHeaders(IEnumerable<KeyValuePair<string, string>> headers)
+		{
+			SetDefaultHeaders();
```

Comment:
Although I understand your use case (the need to add multiple headers at once). Step away from that for a second and it looks like this function (by it's signature) would actually replace any existing headers with the set you put in. Perhaps another maybe optional parameter to control whether you are setting them additively or replacing all.

It also brings up another question about the usage of `SetDefaultHeaders` If you are just additively setting all the headers you passed in, do we even need that, They will already be there right ?

Reply:
Unless I'm missing something it will perform as you expected (replacing all the headers), it replaces any headers with default headers before adding the new ones.

I'd actually think that in most cases you need to set a lot of pre-defined headers at the time you create the request. Why not add an overloaded constructor with headers to be merged with the defaults?

Also I don't think default headers should ever be called outside of the constructor, its weird that I can manipulate the headers and/or super headers and then the defaults get reapplied.

In fact as I'm writing the more I realise that the super headers are a bad pattern, a singleton that can change what I expect my default headers to be at any time? Is just weird interdependent state, also replacing the word *global* with *super* does not make this inheritance as one might expect.

Reply:
I've made some changes to address this, mainly due to the fact that SetHeader is additive, and SetHeaders wasn't, which is nonsense - the class now takes an additionalHeaders optional parameter on construction, and the SetHeaders function behaves as you'd expect from an external view of the class - just sets some headers.

The deeper problem with the super-headers design pattern isn't something I think is in the scope of these changes but can take them on in a subsequent branch if original authors (i.e. Harry, I think?) agree.